ml-algorithms/src/main/java/org/opensearch/ml/engine/memory/MLMemoryManager.java

```diff
@@ -0,0 +1,218 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.ml.engine.memory;
+
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+
+import org.opensearch.action.search.SearchRequest;
+import org.opensearch.action.search.SearchResponse;
+import org.opensearch.action.update.UpdateResponse;
+import org.opensearch.client.Client;
+import org.opensearch.client.Requests;
+import org.opensearch.cluster.service.ClusterService;
+import org.opensearch.common.util.concurrent.ThreadContext;
+import org.opensearch.core.action.ActionListener;
+import org.opensearch.index.query.BoolQueryBuilder;
+import org.opensearch.index.query.ExistsQueryBuilder;
+import org.opensearch.index.query.QueryBuilders;
+import org.opensearch.index.query.TermQueryBuilder;
+import org.opensearch.ml.common.conversation.ConversationalIndexConstants;
+import org.opensearch.ml.common.conversation.Interaction;
+import org.opensearch.ml.memory.action.conversation.CreateConversationAction;
+import org.opensearch.ml.memory.action.conversation.CreateConversationRequest;
+import org.opensearch.ml.memory.action.conversation.CreateConversationResponse;
+import org.opensearch.ml.memory.action.conversation.CreateInteractionAction;
+import org.opensearch.ml.memory.action.conversation.CreateInteractionRequest;
+import org.opensearch.ml.memory.action.conversation.CreateInteractionResponse;
+import org.opensearch.ml.memory.action.conversation.GetInteractionsAction;
+import org.opensearch.ml.memory.action.conversation.GetInteractionsRequest;
+import org.opensearch.ml.memory.action.conversation.GetInteractionsResponse;
+import org.opensearch.ml.memory.action.conversation.UpdateInteractionAction;
+import org.opensearch.ml.memory.action.conversation.UpdateInteractionRequest;
+import org.opensearch.ml.memory.index.ConversationMetaIndex;
+import org.opensearch.search.SearchHit;
+import org.opensearch.search.builder.SearchSourceBuilder;
+import org.opensearch.search.sort.SortOrder;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Preconditions;
+
+import lombok.AllArgsConstructor;
+import lombok.extern.log4j.Log4j2;
+
+/**
+ * Memory manager for Memories. It contains ML memory related operations like create, read interactions etc.
+ */
+@Log4j2
+@AllArgsConstructor
+public class MLMemoryManager {
+
+    private Client client;
+    private ClusterService clusterService;
+    private ConversationMetaIndex conversationMetaIndex;
+    private final String indexName = ConversationalIndexConstants.INTERACTIONS_INDEX_NAME;
```

Comment:
Why wrap another static constant with this final String ?

Reply:
This is removed in the refactor.